crates/air-lib/trace-handler/src/state_automata/fold_fsm/lore_applier.rs

```diff
@@ -44,16 +44,20 @@ fn apply_fold_lore(
     ctx_type: MergeCtxType,
     next_position: ByNextPosition,
 ) -> FSMResult<()> {
-    let fold_lore = match fold_lore {
-        Some(fold_lore) => fold_lore,
-        None => return Ok(()),
-    };
-
     let slider = match ctx_type {
         Previous => data_keeper.prev_slider_mut(),
         Current => data_keeper.current_slider_mut(),
     };
 
+    let fold_lore = match fold_lore {
+        Some(fold_lore) => fold_lore,
+        None => {
+            // create an empty subtrace
+            slider.set_position_and_len(slider.subtrace_len(), 0)?;
```

Comment:
Using `slider.subtrace_len()` for a position is kinda confusing. From the other hand `slider` has a method `set_subtrace_len`, I suggest to use it instead.